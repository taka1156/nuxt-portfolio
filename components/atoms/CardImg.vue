<template functional>
  <img v-lazy="props.cardImg" class="card__img" />
</template>

<style scoped>
.card__img {
  margin: 0 auto;
  height: 100%;
  width: 100%;
}

/* lazy-load */
img[lazy='loaded'] {
  animation: 1s anime-opacity both;
}
</style>
