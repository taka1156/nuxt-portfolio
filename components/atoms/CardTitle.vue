<template functional>
  <h2 class="card__title">{{ props.cardTitle }}</h2>
</template>

<style scoped>
.card__title {
  font-size: 1.75em;
  line-height: 0em;
  text-align: center;
}
</style>
