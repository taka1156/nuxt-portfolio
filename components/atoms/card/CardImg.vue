<template functional>
  <img
    v-lazy="props.cardImg"
    :alt="props.imgAlt"
    class="card-img"
    :class="[props.isPortfolio ? 'card-img--lg' : 'card-img--sm']"
  />
</template>

<style scoped>
.card-img {
  display: block;
  margin: 0 auto;
  padding: 5px;
  box-sizing: border-box;
}

.card-img--sm {
  height: 50%;
  width: 50%;
}

.card-img--lg {
  height: 100%;
  width: 100%;
}
</style>
