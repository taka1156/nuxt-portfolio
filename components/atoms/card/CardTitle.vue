<template>
  <div>
    <h2 class="card-title">
      <!-- @slot カードタイトル -->
      <slot />
    </h2>
  </div>
</template>

<style scoped>
.card-title {
  font-size: 1.3em;
  line-height: 0em;
  text-align: center;
}
</style>
