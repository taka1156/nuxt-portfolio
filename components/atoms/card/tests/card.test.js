import { shallowMount } from '@vue/test-utils';
import CardTitle from '../CardTitle.vue';
import CardImg from '../CardImg.vue';

describe('CardImg', () => {
  it('値がDOMに反映されているか', () => {
    const dummyImg = {
      img: 'http://placehold.jp/150x150.png',
      alt: 'ダミー画像',
    };
    const navImg = shallowMount(CardImg, {
      propsData: {
        imgUrl: dummyImg.img,
        imgAlt: dummyImg.alt,
        size: 'lg', //ナビゲーションはlgで固定
      },
    });
    const imgTag = navImg.find('img');
    // v-lazyからsrcを取り出せない
    // expect(imgTag.attributes().src).toBe(dummyImg.img);
    expect(imgTag.attributes().alt).toBe(dummyImg.alt);
    expect(navImg.html()).toMatchSnapshot();
  });
});

describe('CardTitle', () => {
  it('値がDOMに反映されているか', () => {
    const dummyTitle = 'ダミータイトル';
    const cardTitle = shallowMount(CardTitle, {
      slots: {
        default: dummyTitle,
      },
    });
    const hTag = cardTitle.find('h2');
    expect(hTag.text()).toBe(dummyTitle);
    expect(cardTitle.html()).toMatchSnapshot();
  });
});
