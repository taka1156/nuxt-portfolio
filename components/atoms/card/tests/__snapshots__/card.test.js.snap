// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImg 値がDOMに反映されているか 1`] = `"<div><img alt=\\"ダミー画像\\" class=\\"card-img card-img--lg\\"></div>"`;

exports[`CardTitle 値がDOMに反映されているか 1`] = `
"<div>
  <h2 class=\\"card-title\\">ダミータイトル</h2>
</div>"
`;
